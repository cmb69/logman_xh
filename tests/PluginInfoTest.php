<?php

namespace Logman;

use ApprovalTests\Approvals;
use Logman\Infra\View;
use PHPUnit\Framework\TestCase;

class PluginInfoTest extends TestCase
{
    public function testDisplaysPluginInfo(): void
    {
        $view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["logman"]);
        $sut = $this->getMockBuilder(PluginInfo::class)
            ->setConstructorArgs(["./", $view])
            ->onlyMethods(["isXhVersionAtLeast", "isPhpVersionAtLeast", "isWritable"])
            ->getMock();
            $sut->expects($this->any())->method("isXhVersionAtLeast")->willReturn(true);
            $sut->expects($this->any())->method("isPhpVersionAtLeast")->willReturn(true);
            $sut->expects($this->any())->method("isWritable")->willReturn(true);
            Approvals::verifyHtml($sut("en"));
    }
}
