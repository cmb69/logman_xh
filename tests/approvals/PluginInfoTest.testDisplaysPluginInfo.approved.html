
<h1>Logman 1.1-dev</h1>
<div class="logman_syscheck">
  <h2>System Check</h2>
  <p class="xh_success">CMSimple_XH version is at least 1.7.0 – good</p>
  <p class="xh_success">PHP version is at least 7.4.0 – good</p>
  <p class="xh_success">./config/config.php is writable – good</p>
  <p class="xh_success">./css/stylesheet.css is writable – good</p>
  <p class="xh_success">./languages/en.php is writable – good</p>
</div>
